File: src/main/java/com/github/svegon/capi/event/render/tooltip/EnderChestTooltipCallback.java
package com.github.svegon.capi.event.render.tooltip;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.client.item.TooltipContext;
import net.minecraft.item.ItemStack;
import net.minecraft.text.Text;
import net.minecraft.world.BlockView;
import org.jetbrains.annotations.Nullable;

import java.util.List;

@FunctionalInterface
public interface EnderChestTooltipCallback {
    Event<EnderChestTooltipCallback> EVENT = EventFactory.createArrayBacked(EnderChestTooltipCallback.class,
            (stack, world, tooltip, options) -> {}, listeners -> (stack, world, tooltip, context) -> {
        for (EnderChestTooltipCallback listener : listeners) {
            listener.appendEnderChestTooltip(stack, world, tooltip, context);
        }
    });

    void appendEnderChestTooltip(ItemStack stack, @Nullable BlockView world, List<Text> tooltip, TooltipContext options);
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/render/tooltip/EnderChestTooltipCallback.kt
package io.github.svegon.capi.event.render.tooltip

import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.client.item.TooltipContext
import net.minecraft.text.Text
import net.minecraft.world.BlockView

fun interface EnderChestTooltipCallback {
    fun appendEnderChestTooltip(stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, options: TooltipContext?)

    companion object {
        @JvmField
        val EVENT: Event<EnderChestTooltipCallback> = EventFactory.createArrayBacked<EnderChestTooltipCallback>(
            EnderChestTooltipCallback::class.java,
            EnderChestTooltipCallback { stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, options: TooltipContext? -> }) { listeners: Array<EnderChestTooltipCallback> ->
            EnderChestTooltipCallback { stack: ItemStack?, world: BlockView?, tooltip: List<Text?>?, context: TooltipContext? ->
                for (listener in listeners) {
                    listener.appendEnderChestTooltip(stack, world, tooltip, context)
                }
            }
        }
    }
}
